### PR TITLE
fix(worker): coordinate repositories independently

### LINKED ISSUE #202: Remove worker-wide repository preparation serialization
## Outcome

Attempts for unrelated repositories can prepare concurrently, while operations that mutate one repository's shared Git metadata remain serialized. A slow clone or fetch must not consume the worker pool behind one global repository lock.

## Context

The worker already runs several attempts concurrently, and issue #201 raises the default pool size to 10. Managed repository acquisition currently holds one worker-wide `repositoryCacheMutex` across cache inspection, clone, repository validation, remote lookup, fetch, and base-commit resolution. Several preparing attempts can therefore occupy every worker slot while waiting behind one slow repository operation, even when they target unrelated repositories.

Attempts need isolated runtime execution, but Git worktrees still share a repository common directory. The locking boundary should protect shared Git metadata briefly without serializing the agent subprocess.

## Constraints

- Replace the worker-wide acquisition lock with keyed per-repository coordination.
- Use the control-plane repository ID for managed repositories and the canonical repository path or normalized remote identity for legacy repositories.
- Serialize clone installation, base resolution and fetch, worktree add/remove, and managed branch cleanup for the same repository where they mutate shared Git metadata.
- Never hold a repository lock while Codex or Claude Code is running.
- Unrelated repositories must not block each other's clone, fetch, worktree creation, cleanup, or runtime execution.
- Preserve the 100-entry managed repository cache limit under concurrent first-time clones. Reserve cache capacity atomically before releasing any short global cache-accounting lock.
- Two concurrent first-time requests for the same repository must install exactly one valid cache entry.
- Failed, cancelled, and timed-out preparation must release its repository lock and any cache reservation.
- Bound or remove idle keyed-lock entries so a long-running worker cannot grow the coordination map without limit.
- Preserve origin revalidation, exact base-commit freezing, manifest durability, fail-closed cleanup, and retained-worktree accounting.

## Acceptance criteria

- A deliberately blocked clone for repository A does not prevent an already cached task for repository B from reaching its runtime.
- First-time clones for repositories A and B can overlap.
- Two concurrent claims for the same uncached repository perform one clone installation and both receive a valid resolved repository.
- Two tasks for the same repository can run concurrently in distinct worktrees after their short Git metadata operations complete.
- Concurrent worktree creation and cleanup for one repository do not leave partial paths, duplicate registrations, missing branches, or an unhealthy worker.
- Concurrent cache reservations never exceed the cache-entry limit.
- Preparation failure or cancellation cannot leak a keyed lock or cache reservation.
- The race detector reports no access races in repository acquisition, worktree lifecycle, or lock bookkeeping.
- Worker documentation describes which preparation operations serialize and confirms that runtime sessions remain parallel.

## Checks

- `go test ./internal/worker`
- `go test -race ./internal/worker`
- Add barrier-based tests for unrelated slow clone versus cached work, overlapping first-time clones, duplicate same-repository acquisition, same-repository runtime overlap, and concurrent cleanup.
- Run the complete repository test suite after the focused checks.

## Dependencies

- This issue can be implemented independently of #201, but it should be complete before relying on high worker capacities in normal local use.

## Out of scope

- Automatic cache eviction.
- Sharing one repository cache between different worker identities.
- Running agent subprocesses inside a stronger filesystem or network sandbox.
- Changing control-plane task routing or lease semantics.

### PR DESCRIPTION
**Pull request by owainlewis**
Closes #202

## Summary
- replace the worker-wide repository cache mutex with bounded, context-aware per-repository coordination
- reserve managed cache capacity atomically and preserve installed-entry accounting across failures
- serialize same-repository Git metadata operations while allowing runtime sessions and unrelated repositories to overlap
- add barrier-based concurrency, cancellation, cleanup, capacity, and race coverage
- document the repository preparation lock boundary

## Verification
- go test ./internal/worker -count=1
- go test -race ./internal/worker -count=1
- go test ./...
- go vet ./...
- just format-check
- just boundary
- just ui-check
- just test-tooling
- just test-launcher
- independent subagent review: no remaining findings

### CODE REVIEW

File: internal/worker/reconcile.go [hunk 1]
```diff
@@ -499,6 +499,11 @@ func (manager *Manager) cleanCompletedWorktree(attemptID string) error {
 	}
 	ctx, cancel := context.WithTimeout(context.Background(), 2*gitCommandTimeout)
 	defer cancel()
+	releaseRepository, err := manager.repositoryLocks.acquire(ctx, manager.coordinationKeyForManifest(manifest))
```
[the remark this answers is not in the document]
Comment (owainlewis) [reply internal/worker/reconcile.go]: Fixed in fc3d8b9. Automatic cleanup and retention now wait for the repository lock with the five-minute acquisition budget, then start a fresh 60-second Git maintenance context only after acquiring the lock. Focused repository, cleanup, and retention worker tests pass.

File: internal/worker/repository_cache.go [hunk 2]
```diff
@@ -79,26 +79,33 @@ func (manager *Manager) acquireManagedRepository(
...
 	}
 
-	manager.repositoryCacheMutex.Lock()
-	defer manager.repositoryCacheMutex.Unlock()
-
 	acquisitionContext, cancel := context.WithTimeout(ctx, repositoryAcquisitionTimeout)
 	defer cancel()
+	release, err := manager.repositoryLocks.acquire(
+		acquisitionContext, managedRepositoryCoordinationKey(claimRepository.ID),
+	)
```
[the remark this answers is not in the document]
Comment (owainlewis) [reply internal/worker/repository_cache.go]: Fixed in df1d3b9. Managed acquisition now uses one bounded context for waiting on the repository lock and creates a fresh five-minute acquisition context only after the lock is held. Verified with the focused managed repository tests.
